src/Compilers/Core/Portable/PEWriter/PeWriter.cs

```diff
@@ -186,7 +186,7 @@ internal static class PeWriter
                 debugDirectoryBuilder = null;
             }
 
-            var peBuilder = new ManagedPEBuilder(
+            var peBuilder = new ExtendedPEBuilder(
```

Comment:
>ExtendedPEBuilder [](start = 32, length = 17)

Why not have a separate ReferenceAssemblyBuilder and only create it when emitting ref assembly? Instead of heaving a more generic builder whose behavior differs based on a boolean